File: data_structures/market_shock.py
class MarketShock:
    """Event representing an abrupt token price change."""

    def __init__(self, step: int, severity: float) -> None:
        self.step = step
        self.severity = severity  # +/- 0.x multiplier applied to the price

    def to_dict(self) -> dict:
        return {"step": self.step, "severity": self.severity}

    @classmethod
    def from_dict(cls, data: dict):
        return cls(data.get("step", 0), data.get("severity", 0.0))

File: lib/data-structures/market-shock.ts
// Market Shock - abrupt token price change events
// Port from data_structures/market_shock.py

export class MarketShock {
  step: number;
  severity: number; // +/- multiplier applied to price

  constructor(step: number, severity: number) {
    this.step = step;
    this.severity = severity;
  }

  toDict(): { step: number; severity: number } {
    return {
      step: this.step,
      severity: this.severity,
    };
  }

  static fromDict(data: { step?: number; severity?: number }): MarketShock {
    return new MarketShock(
      data.step || 0,
      data.severity || 0.0
    );
  }
}
